ra/ra.go

```diff
@@ -1703,6 +1723,11 @@ func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.New
 	// Otherwise we were unable to find an order to reuse, continue creating a new
 	// order
 
+	// Check if there is rate limit space for a new order within the current window
+	if err := ra.checkNewOrdersPerAccountLimit(ctx, *order.RegistrationID); err != nil {
```

Comment:
I know this is for an API that isn't yet prod but given if this doesn't work how we want (or has negative side effects) I kind of think this should be gated behind a feature... I could probably be convinced that isn't needed though.

Reply:
I think this feedback and your feedback about killing off `checkPendingOrderLimit` are hard to achieve together.

If I delete the `checkPendingOrderLimit` impl in the RA (and the call from `ra.NewOrder` that is already deleted stays deleted), then the feature flag would make `NewOrder` limitless (minus the authz limits ofc) if disabled which seems more harmful overall.

I'm slightly inclined to not introduce the feature flag here.

Reply:
I think it could be done (i.e. `if featureEnabled { limitA } else { limitB }`) but your right it would require keeping the old code around. I'm fine going forward without it to be honest, I'm just hesitant that if we ever need to back it out we'll need to essentially revert and re-deploy, which is a pain.

That said... this code looks right enough to me, so I'm happy to just dive in head first 😄.

Reply:
YOLO

Reply:
>  I'm just hesitant that if we ever need to back it out we'll need to essentially revert and re-deploy, which is a pain.

That's true, but the pain of doing so only in staging is relatively minor compared to a prod revert.

Reply:
For rate limits, the `limit.Enabled()` flag acts as a good feature flag, since `Enabled()` will return false if there is not entry for that limit.